### PR TITLE
Dev frontend

### PR DESCRIPTION
**Pull request by ows5ut**
Little fixes

### CODE REVIEW

**Comment by ows5ut** [comment]
little fixes

File: frontend/src/Components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+type Props = {
+    small?: boolean,
+    medium?: boolean,
+    large?: boolean,
+    color?: string,
+    outline?: boolean
+    circular?: boolean
+}
+
+const Badge:React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.tsx:13]: why do we need small, medium, large - why not pass size?

File: frontend/src/Components/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        children 
+    }) => {
+    let size = 0;
+    if (small) size = 1;
+    if (medium) size = 2;
+    if (large) size = 3;
+    return <span 
+            className={`badge Badge
+            ${className}
+            ${circular ? `circular w-h-${size}` : `px-1 py-${size}`}
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.tsx:30]: what is the difference between Badge with class circular and CircularBadge

File: frontend/src/Components/Badge/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            ${outline ? `text-${color} border border-${color}` :
+            `badge-${color}`}`}>
+                {children}
+            </span>
+}
+
+export const CircularBadge:React.FC<Props & React.HTMLAttributes<HTMLElement>> = (props) => 
+{
+    return <Badge {...props} circular className={`${props.className}  rounded-circle`}/>
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.tsx:39]: remove space `}  r`

File: frontend/src/Components/Badge/Badge.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export default Badge;
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.tsx:42]: new line

File: frontend/src/Components/Badge/index.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import Badge, { CircularBadge as CB} from './Badge';
```
Comment (livnekes) [inline frontend/src/Components/Badge/index.tsx:1]: don't like the CircularBadge as CB - do we need this? can't we do something else or change name if not possible.

File: frontend/src/Components/Breadcrumbs/Breadcrumbs.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.Breadcrumbs {
+    position: relative
+}
```
Comment (livnekes) [inline frontend/src/Components/Breadcrumbs/Breadcrumbs.scss:3]: newline

File: frontend/src/Components/Breadcrumbs/Breadcrumbs.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.Breadcrumbs {
```
Comment (livnekes) [inline frontend/src/Components/Breadcrumbs/Breadcrumbs.scss:1]: why do we need this css

File: frontend/README.md [hunk 8]
```diff
@@ -41,4 +41,4 @@ You don’t have to ever use `eject`. The curated feature set is suitable for sm
 
 You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
```
Comment (livnekes) [inline frontend/README.md:42]: Change readme - not create react app.

File: frontend/package.json [hunk 9]
```diff
@@ -3,26 +3,42 @@
...
   "private": true,
   "dependencies": {
+    "@material-ui/core": "^4.11.0",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.56",
+    "@material-ui/styles": "^4.10.0",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "@types/classnames": "^2.2.10",
```
Comment (livnekes) [inline frontend/package.json:13]: @types should be in devDependencies

File: frontend/postcss.config.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+const tailwindcss = require('tailwindcss');
+module.exports = {
+    plugins: [
+        tailwindcss('./tailwind.js'),
+        require('autoprefixer')
+    ],
+};
```
Comment (livnekes) [inline frontend/postcss.config.js:7]: newline

File: frontend/postcss.config.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const tailwindcss = require('tailwindcss');
+module.exports = {
+    plugins: [
+        tailwindcss('./tailwind.js'),
+        require('autoprefixer')
```
Comment (livnekes) [inline frontend/postcss.config.js:5]: const = require()...

File: frontend/public/index.html [hunk 12]
```diff
@@ -1,21 +1,19 @@
...
-    />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-    <!--
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description" content="Web site created using create-react-app" />
```
Comment (livnekes) [inline frontend/public/index.html:9]: remove create-react-app from all code

File: frontend/public/index.html [hunk 13]
```diff
@@ -1,21 +1,19 @@
...
+  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (livnekes) [inline frontend/public/index.html:10]: logo192 is not a good name.

File: frontend/public/index.html [hunk 14]
```diff
@@ -39,5 +38,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-  </body>
-</html>
+</body>
+
+</html>
```
Comment (livnekes) [inline frontend/public/index.html:43]: newline

File: frontend/src/App.tsx [hunk 15]
```diff
@@ -4,7 +4,7 @@ import './App.scss';
 
 const App: React.FC = () => {
   return (
-    <div className="app">
+    <div className="app" >
```
Comment (livnekes) [inline frontend/src/App.tsx:7]: remove space

File: frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx [hunk 16]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react'
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import { Paper } from '@material-ui/core';
+import { IActivitiesComponent } from './type';
```
Comment (livnekes) [inline frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx:4]: why do we need this IActivitiesComponent?

File: frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react'
```
Comment (livnekes) [inline frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx:1]: why sometimes `import * as React from 'react'` and sometimes `import React from 'react'`

File: frontend/src/Components/Alert/Alert.scss [hunk 18]
```diff
@@ -0,0 +1,17 @@
+@import "../../Pages/MainLayout/themes.scss";
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.scss:1]: choose " or ' and use it everywhere

File: frontend/src/Components/Alert/Alert.scss [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+.Alert {
+    position: relative;
+    &.raised,
+    &.outlined, 
+    &.alert-border-left {
+        background-color: #fff;
+        @include themify($themes) {
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.scss:8]: I don't like this mixin - u don't understand what it does.

File: frontend/src/Components/Alert/Alert.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import './Alert.scss';
+import { Alert as A } from 'reactstrap';
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:3]: I prefer `import { Alert as AlertReactstrap } from 'reactstrap';`

File: frontend/src/Components/Alert/Alert.tsx [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+import NotificationContext, { NotificationContextProvider, NotificationContextType } from '../../Pages/MainLayout/NotificationContext';
+
+type Props = {
+    notification_key: string,
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:7]: use camelCase here

File: frontend/src/Components/Alert/Alert.tsx [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+    open?: boolean,
+    raised?: boolean,
+    borderLeft?: boolean,
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:10]: why do we need borderLeft? what is raised?

File: frontend/src/Components/Alert/Alert.tsx [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+    outlined?: boolean,
+    color?: string
+} & React.HTMLAttributes<HTMLElement>
+
+const Alert: React.FC<Props> = (
+    {
+        notification_key,
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:17]: camelCase

File: frontend/src/Components/Alert/Alert.tsx [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+    raised && (className += `shadow border-0 text-${color} raised`);
+    outlined && (className += `text-${color} border border-${color} outlined`);
+    borderLeft && (className += `rounded-0 shadow-sm border-top-0 border-right-0 border-bottom-0 border-left border-${color} alert-border-left `);
+    return <A color={color} className={`Alert w-100 ${className}`} isOpen={open} toggle={onDismiss} style={{ ...style }}>
+        {children}
+    </A>
+}
+
+export const TopAlert = (props: Props) => {
+    const context = React.useContext<NotificationContextType>(NotificationContext);
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:42]: extract to a method

File: frontend/src/Components/Alert/Alert.tsx [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+    }
+) => {
+    const context = React.useContext<NotificationContextType>(NotificationContext);
+
+    const onDismiss = (event: React.MouseEvent<any, MouseEvent>) => {
+        context.deleteNotification(notification_key);
+    }
+    raised && (className += `shadow border-0 text-${color} raised`);
+    outlined && (className += `text-${color} border border-${color} outlined`);
+    borderLeft && (className += `rounded-0 shadow-sm border-top-0 border-right-0 border-bottom-0 border-left border-${color} alert-border-left `);
```
Comment (livnekes) [inline frontend/src/Components/Alert/Alert.tsx:35]: don't understand all this classes...

File: frontend/src/Components/AuthComponent/AuthComponent.tsx [hunk 26]
```diff
@@ -1,29 +1,28 @@
 import * as React from 'react';
 import InputElement from '../InputElement';
 import { Col, Row, Button, Form, FormGroup, Label, Input } from 'reactstrap';
-import {propsType, stateType} from './types';
+import {propsType, stateType, IField} from './types';
```
Comment (livnekes) [inline frontend/src/Components/AuthComponent/AuthComponent.tsx:4]: use `{ props` and not `{props`

File: frontend/src/Components/AuthComponent/AuthComponent.tsx [hunk 27]
```diff
@@ -1,29 +1,28 @@
 import * as React from 'react';
 import InputElement from '../InputElement';
 import { Col, Row, Button, Form, FormGroup, Label, Input } from 'reactstrap';
-import {propsType, stateType} from './types';
```
Comment (livnekes) [inline frontend/src/Components/AuthComponent/AuthComponent.tsx:4]: why do we need this `IField`

File: frontend/src/Components/Badge/Badge.scss [hunk 28]
```diff
@@ -0,0 +1,23 @@
+.Badge {
+    font-size: .75rem !important;
+    &.circular{
+        font-size: .7rem !important;
+        display: inline-flex;
+        align-items: center;
+        justify-content: center;
+    }
+    &.w-h{
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.scss:9]: not good class names.

File: frontend/src/Components/Badge/Badge.scss [hunk 29]
```diff
@@ -0,0 +1,23 @@
+.Badge {
+    font-size: .75rem !important;
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.scss:2]: why do we need to use !important;

File: frontend/src/Components/Badge/Badge.scss [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+    &.circular{
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.scss:3]: add spaces to all `{` like `&.circular {`

File: frontend/src/Components/Badge/Badge.scss [hunk 31]
```diff
@@ -0,0 +1,23 @@
+.Badge {
```
Comment (livnekes) [inline frontend/src/Components/Badge/Badge.scss:1]: use dash-case (no capital letters)

File: frontend/src/Components/Breadcrumbs/Breadcrumbs.tsx [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+import HomeIcon from '@material-ui/icons/Home';
+
+type LinkType = {
+    href?: string,
+    label?: string,
+}
+
+type Props = {
+    separator?: JSX.Element | string
+    home?: boolean
```
Comment (livnekes) [inline frontend/src/Components/Breadcrumbs/Breadcrumbs.tsx:13]: pass icon from outside and don't use home?

File: frontend/src/Components/Breadcrumbs/Breadcrumbs.tsx [hunk 33]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import './Breadcrumbs.scss';
+import { Breadcrumbs as B, Link } from '@material-ui/core';
+import HomeIcon from '@material-ui/icons/Home';
```
Comment (livnekes) [inline frontend/src/Components/Breadcrumbs/Breadcrumbs.tsx:4]: remove this - pass from outside

File: frontend/src/Components/Button/Button.scss [hunk 34]
```diff
@@ -0,0 +1,3 @@
+.Button {
```
Comment (livnekes) [inline frontend/src/Components/Button/Button.scss:1]: weird style.

File: frontend/src/Components/Button/Button.tsx [hunk 35]
```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import {Button as B} from 'reactstrap';
+import './Button.scss';
+
+type Props = {
+    active?: boolean,
+    outline?: boolean,
+    color?: string,
```
Comment (livnekes) [inline frontend/src/Components/Button/Button.tsx:8]: pass styles instead. check if need this I think

File: frontend/src/Components/Button/Button.tsx [hunk 36]
```diff
@@ -0,0 +1,99 @@
...
+        }} 
+        outline={outline} 
+        color={color}
+        >
+        {children}
+    </B>
+}
+
+type ButtonProps = Pick<Props, Exclude<keyof Props, "outline">>;
+export const Button:React.FC<ButtonProps> = (props) => {
```
Comment (livnekes) [inline frontend/src/Components/Button/Button.tsx:72]: maybe use helper function

File: frontend/src/Components/Button/Button.tsx [hunk 37]
```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import {Button as B} from 'reactstrap';
+import './Button.scss';
+
+type Props = {
+    active?: boolean,
```
Comment (livnekes) [inline frontend/src/Components/Button/Button.tsx:6]: why pass active?

File: frontend/src/Components/Button/Button.tsx [hunk 38]
```diff
@@ -0,0 +1,99 @@
...
+    outline?: boolean,
+    color?: string,
+    bgColor?: string,
+    textColor?: string,
+    boxShadow?: string,
+    onHoverStyle?: React.CSSProperties,
```
Comment (livnekes) [inline frontend/src/Components/Button/Button.tsx:12]: I'm not sure we need this. maybe use only css

File: frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx [hunk 39]
```diff
@@ -0,0 +1,28 @@
...
+import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+
+interface IButton {
+    lable: string
+    items: string[]
+}
+
+const ButtonDropDown = (props: IButton) => {
+    const [dropdownOpen, setOpen] = React.useState(false);
+
```
Comment (livnekes) [inline frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx:11]: remove newlines

File: frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx [hunk 40]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+
+interface IButton {
```
Comment (livnekes) [inline frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx:4]: ButtonProps

File: frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx [hunk 41]
```diff
@@ -0,0 +1,28 @@
...
+
+    const toggle = () => setOpen(!dropdownOpen);
+
+    return (
+        <ButtonDropdown isOpen={dropdownOpen} toggle={toggle}>
+            <DropdownToggle caret>
+                {lable}
+            </DropdownToggle>
+            <DropdownMenu>
+                {items.map((el: string, i: number) => <DropdownItem key={i}>{el}</DropdownItem>)}
```
Comment (livnekes) [inline frontend/src/Components/ButtonDropDown/ButtonDropDown.tsx:22]: item, key

File: frontend/src/Components/Charts/BarChart/BarChart.tsx [hunk 42]
```diff
@@ -0,0 +1,51 @@
...
+  const colors = settings.colors || barChartDefaultSettings.colors
+  return (
+    <ResponsiveContainer>
+    <BarChart
+      width={width}
+      height={height}
+      data={data}
+      barSize={settings.barSize}
+      barCategoryGap={settings.barCategoryGap}
+      margin={{
```
Comment (livnekes) [inline frontend/src/Components/Charts/BarChart/BarChart.tsx:24]: why use inline styles.

File: frontend/src/Components/Charts/BarChart/BarChart.tsx [hunk 43]
```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import {
+  BarChart, Bar, Cell, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer,
+} from 'recharts';
+import { barChartDefaultData, BarChartProps, barChartDefaultSettings } from './BarChartDataDefault'
```
Comment (livnekes) [inline frontend/src/Components/Charts/BarChart/BarChart.tsx:5]: why do we have default data here? it should be passed from outside...

File: frontend/src/Components/Charts/BarChart/BarChart.tsx [hunk 44]
```diff
@@ -0,0 +1,51 @@
...
+
+const BarChartComponent: React.FC<BarChartProps> = ({
+  data = barChartDefaultData,
+  settings = barChartDefaultSettings
+}) => {
+  const width = settings.width || barChartDefaultSettings.width
```
Comment (livnekes) [inline frontend/src/Components/Charts/BarChart/BarChart.tsx:11]: get all from outside.

File: frontend/src/Components/Charts/BarChart/BarChartDataDefault.tsx [hunk 45]
```diff
@@ -0,0 +1,55 @@
+export type BarData =
```
Comment (livnekes) [inline frontend/src/Components/Charts/BarChart/BarChartDataDefault.tsx:1]: delete this file.

File: frontend/src/Components/Charts/LineChart/LineChart.tsx [hunk 46]
```diff
@@ -0,0 +1,41 @@
...
+        ? <CartesianGrid strokeDasharray="3 3" />
+        : null
+      }
+      <XAxis dataKey="category" type={xType} allowDuplicatedCategory={false} axisLine={false} tickLine={false} />
+      <YAxis dataKey="value" axisLine={false} tickLine={false} />
+      <Tooltip />
+      <Legend verticalAlign="top" />
+      
+      {data.map((line, i) => (
+        <Line strokeWidth={chartLine?.strokeWidth} legendType="circle" type={chartLine?.type} stroke={colors[i % colors.length]} dataKey="value" data={line.data} name={line.name} key={line.name} activeDot={chartLine?.activeDot} />
```
Comment (livnekes) [inline frontend/src/Components/Charts/LineChart/LineChart.tsx:34]: format this line.

File: frontend/src/Components/Charts/LineChart/LineChart.tsx [hunk 47]
```diff
@@ -0,0 +1,41 @@
...
+import {
+  LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ReferenceLine, ResponsiveContainer,
+} from 'recharts';
+import { lineChartDataDefault, lineChartDefaultSettings, LineChartProps } from './lineChartDataDefault'
+
+const LineChartComponent: React.FC<LineChartProps> = ({
+  data = lineChartDataDefault,
+  settings = lineChartDefaultSettings
+}) => {
+  const width = settings.width || lineChartDefaultSettings.width
```
Comment (livnekes) [inline frontend/src/Components/Charts/LineChart/LineChart.tsx:11]: don't use default data here.

File: frontend/src/Components/Charts/PieChart/PieChartDataDefault.tsx [hunk 48]
```diff
@@ -0,0 +1,47 @@
+export type PieData =
```
Comment (livnekes) [inline frontend/src/Components/Charts/PieChart/PieChartDataDefault.tsx:1]: extract types & interface to different file.

File: frontend/src/Components/Charts/ScatterChart/ScatterChart.tsx [hunk 49]
```diff
@@ -0,0 +1,39 @@
...
+} from 'recharts';
+import { scatterChartDataDefault, scatterChartDefaultSettings, ScatterChartProps } from './ScatterChartDataDefault'
+import CustomizedDot, {CustomizedLegend} from './ScatterChartDot'
+
+const LineChartComponent: React.FC<ScatterChartProps> = ({
+  data = scatterChartDataDefault,
+  settings = scatterChartDefaultSettings
+}) => {
+  const width = settings.width || scatterChartDefaultSettings.width
+  const height = settings.height || scatterChartDefaultSettings.height
```
Comment (livnekes) [inline frontend/src/Components/Charts/ScatterChart/ScatterChart.tsx:13]: remove defaults

File: frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx [hunk 50]
```diff
@@ -0,0 +1,54 @@
...
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      borderRadius: '50%',
+      color: 'white',
+      height: '25px',
+      width: '25px'
+    },
+    content: {
+      margin: '0  0 1rem 1rem'
```
Comment (livnekes) [inline frontend/src/Components/ActivitiesComponent/ActivitiesComponent.tsx:24]: extra space

**Pull request closed without merging**